core/src/nucleus/mod.rs

```diff
@@ -366,23 +380,63 @@ fn reduce_validate_entry(
     _context: Arc<Context>,
     state: &mut NucleusState,
     action_wrapper: &ActionWrapper,
-    _action_channel: &Sender<ActionWrapper>,
-    _observer_channel: &Sender<Observer>,
+    action_channel: &Sender<ActionWrapper>,
+    observer_channel: &Sender<Observer>,
 ) {
-    let mut _has_entry_type = false;
-
-    // must have entry_type
-    if let Some(ref dna) = state.dna {
-        let action = action_wrapper.action();
-        let es = unwrap_to!(action => Action::ValidateEntry);
-        if let Some(ref _wasm) =
-            dna.get_validation_bytecode_for_entry_type(&es.zome_name, &es.type_name)
-        {
-            // TODO #61 validate()
-            // Do same thing as Action::ExecuteZomeFunction
-            _has_entry_type = true;
+    let action = action_wrapper.action();
+    let entry = unwrap_to!(action => Action::ValidateEntry);
+    match zome_name_for_entry_type(&state.dna.clone().unwrap(), entry.entry_type()) {
+        None => {
+            let error = format!("Unknown entry type: '{}'", entry.entry_type());
+            state
+                .validation_results
+                .insert(action_wrapper.clone(), Err(error.to_string()));
         }
-    }
+        Some(zome_name) => {
+            state.validations_running.push(action_wrapper.clone());
```

Comment:
`validations_running` doesn't seem to be useful for anything.
Maybe you are planning to use this later to make sure a validation of an entry is not pushed twice?

Reply:
Yeah, you're right, and I was thinking about not committing it.
The main reason I did is to communicate context. This is how, I think, we should think about the state and this pattern.

I would like us to have a way of formatting or even visualizing the state soon, so that we can reason about state changes easily when debugging. This is a first hint to that...

Reply:
I see two possible changes then:
1. Add `#[cfg(debug)]` in front of `validations_running`. So we know its for debugging only.
2. Panic or return error if we add the same validateEntry twice in that vector, to make `validations_running` actually useful.

Reply:
right, thanks. I will `#[cfg(debug)]` then. Ok?

Reply:
👍